### PR TITLE
Helicopter cleanup

### PR DESCRIPTION
**Pull request by derSteFfi**
some Cleanup in Map_Screen_Helicopter. Stuff that I noticed when working on one of the segfaults. Some of this is purely cosmetic, some is refactoring. Shouldn't hurt anybody in either way. I also tried to split it up into well-sorted commits. Comments welcome :)

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
Nice work. Some of them aren't clear cut - did you do any testing too?

**Comment by derSteFfi** [comment]
yes, i flew around based on a savegame from a previous bug. Landed back in Drassen, loaded a merc, unloaded, transported, stuff like this. Not extensive though

File: Build/Strategic/Map_Screen_Helicopter.cc [hunk 1]
```diff
@@ -436,21 +383,14 @@ BOOLEAN IsHelicopterPilotAvailable( void )
...
-		return( FALSE );
-	}
+	if( gMercProfiles[ SKYRIDER ].iBalance < 0 ) return( FALSE );
 
 	// Drassen too disloyal to wanna help player?
-	if ( CheckFact( FACT_LOYALTY_LOW, SKYRIDER ) )
-	{
-		return ( FALSE );
-	}
+	if ( CheckFact( FACT_LOYALTY_LOW, SKYRIDER ) ) return ( FALSE );
```
Comment (selaux) [inline Build/Strategic/Map_Screen_Helicopter.cc]: Couldnt this be `return CheckFact( FACT_LOYALTY_LOW, SKYRIDER )`?

**Comment by selaux** [comment]
Nice work. I didn't do any testing, but codewise a significant improvement.

**Comment by lynxlynxlynx** [comment]
Someone try recruiting and maybe a sam site acquisition and I'll be completely calm.

File: Build/Strategic/Map_Screen_Helicopter.cc (hunk 1 above)
Comment (derSteFfi) [inline Build/Strategic/Map_Screen_Helicopter.cc]: i think if, than it's ```return !CheckFact( FACT_LOYALTY_LOW, SKYRIDER )```. Question is whether this doesn't look too distorted then.

**Comment by derSteFfi** [comment]
coming tomorrow.

File: Build/Strategic/Map_Screen_Helicopter.cc (hunk 1 above)
Comment (lynxlynxlynx) [inline Build/Strategic/Map_Screen_Helicopter.cc]: You're right and it looks readable to me — especially if I close my eyes a bit and ignore the odd spacing (imo).

Comment (derSteFfi) [inline Build/Strategic/Map_Screen_Helicopter.cc]: OK, I'm gonna do that tomorrow too.

**Comment by derSteFfi** [comment]
OK, tested. I was able to recruit Skyrider, fly around after freeing first SAM, tested hostile airspace and detection of enemies in both hostile and freed sectors. Everything works, no chrashes. #251 still happens so as far as ican tell, completly same behaviour :)

**Pull request merged**